lib/data_update_scripts/20201218080343_update_default_email_addresses.rb

```diff
@@ -0,0 +1,17 @@
+module DataUpdateScripts
+  class UpdateDefaultEmailAddresses
+    def run
+      return if SiteConfig.email_addresses[:contact].present?
+
+      # In order to trigger the attribute setter on the SiteConfig hash object,
+      # it seems like we need to set all the values in the hash and not only the one we are changing.
+      SiteConfig.email_addresses = {
```

Comment:
Something else to add here: 

I thought instead of setting the values manually here, I could just merge the hash like this:
```
[100] pry(main)> SiteConfig.email_addresses
  SiteConfig Load (0.5ms)  SELECT var, value FROM "site_configs"
=> {"default"=>"default@dev.to", "business"=>"business@dev.to", "privacy"=>"privacy@dev.to", "members"=>"members@dev.to", "contact"=>"yo@dev.to"}


[101] pry(main)> SiteConfig.email_addresses.merge!(:password => 'topsecret')
=> {"default"=>"default@dev.to", "business"=>"business@dev.to", "privacy"=>"privacy@dev.to", "members"=>"members@dev.to", "contact"=>"yo@dev.to", "password"=>"topsecret"}


[102] pry(main)> reload!
Reloading...
  ProfileField Load (0.4ms)  SELECT "profile_fields".* FROM "profile_fields" ORDER BY "profile_fields"."id" ASC LIMIT $1  [["LIMIT", 1000]]
  User Exists? (0.6ms)  SELECT 1 AS one FROM "users" LIMIT $1  [["LIMIT", 1]]
Unset ENV variable: COMMUNITY_COPYRIGHT_START_YEAR.
Unset ENV variable: STRIPE_SECRET_KEY.
Unset ENV variable: STRIPE_PUBLISHABLE_KEY.
  SiteConfig Load (0.5ms)  SELECT var, value FROM "site_configs"
=> true

[103] pry(main)> SiteConfig.email_addresses
  SiteConfig Load (0.4ms)  SELECT var, value FROM "site_configs"
=> {"default"=>"default@dev.to", "business"=>"business@dev.to", "privacy"=>"privacy@dev.to", "members"=>"members@dev.to", "contact"=>"yo@dev.to"}
[104] pry(main)>
```
But to no avail.

Reply:
Rhymes brought this to my attention, rails-settings-cached since only defines this method for each value which is why the merge doesn't work https://github.com/huacnlee/rails-settings-cached/blob/4143b0ca2b5696f1a2209cd8e992f24e08d8081d/lib/rails-settings/base.rb#L91-L101